hack/verify-license.sh

```diff
@@ -40,6 +40,7 @@ missing_license_header_files="$($ADDLICENSE_BIN \
   -ignore "**/*.yml" \
   -ignore "**/*.json" \
   -ignore ".idea/**" \
+  -ignore ".git/**"
```

Comment:
@tiansuo114 Can you help to recall why need to ignore `.git` here?

PS: I'm exploring the reason why the license check becomes inactive, and seems this line breaks the functionality.

Reply:
In my detailed response under https://github.com/karmada-io/karmada/pull/5984#issuecomment-2562447851, the idea is that files under the `.git/` directory, similar to those under `.github/`, `.idea/`, and other paths, should not be subject to license checks. When I ran the related check scripts locally, I encountered issues related to this, so I added path ignore rules to the script.